tokio/tests/rt_common.rs

```diff
@@ -9,38 +9,41 @@ macro_rules! rt_test {
         mod basic_scheduler {
             $($t)*
 
-            fn rt() -> Runtime {
+            fn rt() -> Arc<Runtime> {
                 tokio::runtime::Builder::new()
                     .basic_scheduler()
                     .enable_all()
                     .build()
                     .unwrap()
+                    .into()
             }
         }
 
         mod threaded_scheduler_4_threads {
             $($t)*
 
-            fn rt() -> Runtime {
+            fn rt() -> Arc<Runtime> {
                 tokio::runtime::Builder::new()
                     .threaded_scheduler()
                     .core_threads(4)
                     .enable_all()
                     .build()
                     .unwrap()
+                    .into()
             }
         }
 
         mod threaded_scheduler_1_thread {
             $($t)*
 
-            fn rt() -> Runtime {
+            fn rt() -> Arc<Runtime> {
```

Comment:
why is this necessary? isn't `Runtime` already `Clone` now, since it has internal `Arc`s? or am i missing something?

Reply:
We removed the internal `Arc` infavor of having it external. All methods take `&self` (except shutdown) and Runtime impls `Sync`.